### PR TITLE
CMSS13 2.0 -The Future and Beyond

### PR DESCRIPTION
**Pull request by blackdragonTOW**
### Intro:
Hello and welcome to the CM13 2.0 release! We’ve settled on the milestone ‘2.0’ due to the volume and scope of the changes involved.  We truly believe that this is the next step for CM as we know it and will bring the game into a bright new future for new and old players alike.  As we know the playerbase can sometimes be against change of any kind: we ask you to go into this with an open mind and look for the good aspects of this as it will create a greatly immersive experience for everyone!
### General Systems:
- View Distances - Largely player views are very limited, we’ve decided to extend this by 50% to add to player Situational Awareness and add extra enjoyment to the game.
- Fun Facts - Fun Facts now have new types of stat beyond kills/damage.  Including: Messes Cleaned, Meals Cooked, Arrests Made, and Commands sent.
- Weather Updates - Storms will now cause lightning strikes.  These will deal 250 Burn to anyone struck.
- Fishing is here!! - Certain maps with water have fish! This will integrate with pre-existing fishing systems and allow marines (and specific Xenos!) to fish!
- TGUI Updates - The following actions have been updated to TGUI: M41 round counter, Egg placement, Weapon Attachments, and Ladders
- Automated AHelps - We’ve integrated ChatGPT into the AHelp system and have found that 90% of the time, any given AHelp can be resolved successfully with the use of the AI system. We’ll be experimenting with it to verify that we can offload a heavy amount of work off to a fully automated system.
- Whitelist Accessibility - All whitelisted roles (Synth/Pred/CO) will have permadeath. Meaning if a whitelisted character dies, they must reapply for the whitelist with a new character concept. 
- Emote Bloat - Given that it has no mechanical advantage, the warcry emote has been removed.
- Sprite Updates - We will be updating sprites to a 16:9 format for a more modern look.
- Latency Issues - The lag has been fixed!
- MP/Community Program - MPs historically have only been in charge of punishments for IC rule breaks.  To help reduce moderator workload, we will now allow MPs to issue actions to players who break  OOC rules, including up to 24hr bans.
- Cross Server Info Sharing - We’ve exchanged CKey records with the admins of S.P.L.U.R.T. to figure out which of you “accidentally” clicked them in the Server Hub.
### Marine Gameplay:
- CPR - To add soul to CPR and increase the flow of marines going to medbay, performing CPR now has a 10% chance to break the ribs.
- Bleeding - Bleeding historically is ignored and instead treated by Bica.  To combat this: Bleeding debuffs will no longer go away, until they are patched with a bandage.
- Adding Thirst - In addition to hunger, we’ve decided to add thirst as a core system for marines.  This will play into many other aspects such as blood generation.  Given that MREs come with a bottle of water, we expect that this shouldn’t be too big of a deal to find water, and should add use to an otherwise ignored item.
- Marine Law Knowledge Requirements - All Officer roles now require 5 hours of MP.  Lack of knowledge of ML is not an excuse.  IOs and POs will probably be the most affected but we believe this will lead to better game outcomes.
- Dig a hole, Fill a hole! - With an entrenching tool, Marines can now fill tunnels back in!
- Marine Running - Adding a human running animation for marines who are moving.
- Role Bloat - After some serious hard looks at what roles add what to the game.  We are removing Riflemen from selectable USCM roles.  Everything that the Rifleman brings to the table is already covered by another class, so it only makes sense that it is superfluous.
- CO Deployment Updates - COs must now fax USCM HC to get permission to deploy.
- Grenade Tossing - Grenade throwing is now a function of how long you hold the throw button. For example: Holding your throw key for one second will throw one tile, holding it for three seconds will throw it three tiles, and so on.  Maximum range of eight tiles based on item weight. Must stand still while throwing.
- Cosmetic Items - We have removed cosmetic items from the preference tab.  Instead, to encourage CL RP marines can now purchase cosmetic items through the CL with cash.
- Engine Maintenance - Maintenance Techs and the Chief Engineer typically have little to do that contribute to the functions of the ship beyond goofy shenanigans.  To facilitate a better play experience, the Fusion Engines at the back of the ship now require regular maintenance.  The multitool will now tell anyone with engineering skill which tool is needed to tune up a given engine, which will require a 30 second channeled interaction to fix.  We roll to see if a tuneup is needed every 5 minutes.  Every unserviced engine that rolls this need a second time in a row will become “Broken Down” and a third in a row will damage it permanently. 
- Marine Boots - Boots have sizes now, for every size off from the Marine’s shoe size (set in preferences) the Marine will incur a stacking 5% slow.  Boot size can be read by inspecting any given pair of boots. 
- Adjustments to Medical - Realistically humans don’t really recover from the worst traumatic injuries like they do in our game. Brain damage is now permanent.   You will be able to remove 33% of it if your CMO has specced into the new “Brain Surgeon” specialization. 
- Warrant Officers - Added Warrant Officer ranks 1-4.  All Department Heads and Flight Officers are now WOs.
- New OB - Rods from God: The AGM-192 Tungsten Penetration Round has the ability to pierce through roof, ceiling, cave and hivecore protected tiles.  The impact space is a 5x5 grid in which anything touched will be gibbed.  Must be directly lazed.
- New OB - Steel Rain: The XM-199 DPICM is a sub-munition scattering weapon that will hang targeted munitions in the air.  As the sub-munitions slowly fall to the ground over 20 seconds, scanners on each sub-munition pick up potential targets and then with a targeted detonation, knock the target flying with massive AP damage and a 6 second stun. (Note: IFF does not apply)
- New OB - Agent Orange 2.0: Deletes all flora within 20 tiles. Marines caught within the blast may have long term consequences.
- Craftables - The Lunge Mine is now craftable with Engineering skill 3
- New Researcher Path -  Clone Geneticist: you can clone marines via the artificial womb program, may select A.W. name on birth.
- Nickname Update - Following the adage you can’t pick your nickname, players no longer can choose their nicknames.  Players now have access to a modified Medals Panel where they can nominate a nickname for a player.  At the end of the round the player may choose to accept the nickname and “lock it in” for future rounds.
- Req Droppods - Marines can now be sent down via Req droppod.
- ARES Playable - ARES is now a playable role.
- New Rifle - M41 Mk2 Bullpup.
- Barricades - In the interest of simplifying FOB construction, all barricades are now folding barricades.  Folding barricades will now also automatically link together, saving time.
- CL Briefcase Update - Added Cocaine to W-Y briefcase.
- CL Computer Update - Now automatically hooks into the large Fax and Forms google doc, allowing players to select and print them out via a search bar, that way players can actually use the huge 100 form long list available to them.
- New SADAR Missile - Nikita: HE missile that is controllable as a vehicle when fired. Lasts 20 sec or until it impacts any solid object.
- Autopsies - To safeguard against rounds where all IOs die in the first five minutes: successfully performing autopsies on a dead person now yields 1 Intel Point.
- UNGA - UNGA formula now needs to be researched and is randomized every round.
- Meal Service - Mess Techs now must serve breakfast/lunch/dinner at set times throughout the round (currently breakfast is set to end at 00:30, lunch at 01:00 and dinner at 01:30)
- Smartgun IFF - SG historically have been hiding behind other squadmates, keeping them out of danger.  We feel that SG gameplay is more rewarding when you’re directly in danger.  Removing SG IFF will move SG players back to where they should be.
- Smartgun Bolt Maintenance - Given the positive response to Smartgun jamming and dust cover reloads from the community, we’re introducing an additional point of failure to the SG for better soul.  Each time that the SG jams, there is now also a 1:5 chance for the bolt to lock as well.  Clearing a locked bolt is done with the following simple procedure: Click the SG with an open hand to remove the magazine, Alt-click the SG to open the dust cover, Click the SG with an open hand to rack the bolt back into the ‘open position’, with a screwdriver and on ‘grab’ intent click on the bolt section of the SG sprite to pry the bolt forward, with an open hand click on the bolt on the SG sprite to move it to your opposite hand, remove your SG bolt maintenance toolkit from your SG belt (this new item will take up one slot in your belt), remove the bolt wrench and click the bolt once to set it into an unlocked position, use the gun lubricant item from your kit on the bolt to lubricate the part, use the bolt wrench a second time to lock the bolt back into a locked position (if it does not go into locked, try on harm intent to force it), with an open hand click the SG to pull the bolt carrier back, select the bolt and place it in your open hand (If you drop it on the ground the bolt with get dirty and you will have to clean and re-lubricate the part), Click the SG bolt carrier with the bolt (MUST BE ON HELP INTENT OR YOU MAY BREAK YOUR SG PERMANENTLY), once placed remove the headspace timing gauge from your SG maintenance kit, On the headspace timing gauge select the 1/16th setting, on help intent (harm intent may force the gauge further than intended) click the SG to check the GO reading on your headspace timing gauge, if the UI tells you that the GO gauge does NOT fit click the barrel of the SG to rotate once (there will be a click to verify that it has moved), repeat this process until the headspace timing gauge reads GO, click on the timing gauge to turn the GO side to the NOGO side, click the SG again with the timing gauge, if the NOGO side does NOT go in you have appropriately set headspacing, if the NOGO side DOES go in you should restart this process from the beginning. Now we check timing, place the headspace timing gauge into the weapon again, attempt to fire the SG while the gauge is inserted (it should not fire), remove the trigger assembly using a screwdriver click on the SG, click the assembly with an open hand, reattach the assembly to the SG, reinsert the gauge into the SG, repeat the firing check, if done correctly you will hear the bolt click in an attempt to fire, if not, repeat the above assembly removal process again until it does fire.  Once the SG dry fires remove the headspace timing gauge from the SG with an open hand.  With an open hand click the SG once to rack the weapon into the firing position, insert the headspace timing gauge on the NOGO side to verify that the weapon does not fire, remove the headspace timing gauge with an open hand, change the gauge to the GO side, reinsert the gauge again on the GO setting, attempt to fire the weapon (you should receive a dry fire click), remove the headspace timing gauge, reinsert a magazine, click the dust cover to close it, open hand click the SG twice to prep the weapon, you are now ready to fire again.   We hope that this straightforward and intuitive process brings more immersion and soul to the SG role.
### Xeno Gameplay:
- New Xeno Whitelists - Given the importance of T3 strains for a successful hive, we are moving T3s to a WL.  We’ll be pulling data on players’ play times and the top 6 players with the most playtime on each of the T3 strains will become the WL Senators for their respective strains.
- Drone Drool - As seen in the Alien franchise, xenos are constantly dripping with saliva.  Drones will now trail spit as they walk around and lurk in areas, sticking to the ground like blood, that can be followed. 
- Drone Encouragement - The primary fantasy of what a xeno hive looks like consists of many drones and fewer specialists.  In order to facilitate this and bring the game in line with the original Alien movie as well as Alien Isolation, we are rebalancing the Drone strain by giving it a flat 30 armor.
- Nest Resist - Being trapped is not fun.  Now, much like handcuffs, you can escape being nested by resisting with a 1 minute channeled resist.
- Adding VOIP - We have added a new VOIP system to all xeno players to represent hivemind. Given that we feel that xenos would be connected at all times we have set the default setting to Open Mic to allow real connection with your fellow xeno players.
- Friendly Fire - Historically marine players account for a significant amount of their own deaths. They are also more cautious than xeno players for this reason.  With that in mind we will be introducing  Friendly Fire to the xeno section of the game for all ranged attacks. (Yes, including Boiler)
- Xeno Law -  Xeno RP standards have historically been lax.  To tighten that up we are introducing “Xeno Law” as a way to help keep RP in line.  Additionally we will be adding an MP analog that will keep the hive “In Check”.  The Queen Enforcer strain will enforce xeno law to make sure xeno RP standards are held to what they should be, and will be answerable to the QM.
- Larva Growth - To add soul to the larva stage, all larva must now eat in order to grow to T1.  This can be any corpse, small animal, or human. 
- Hats! - Add Xeno cosmetic hats, will share colors with xeno backpacks.
- Pyrophobia - When on fire xenos will now move in a randomized direction until the fire is extinguished.
- New Strain -  Regen melee strain - Vampire Ravager
- New Strain - Fire based range strain - Broiler
- Boiler Update - New ability: self destruct
- Runner Speed - We feel the strain does not align with what we want it to play like.  Increased runner speed: 35% Additionally, Runners will have momentum based on run speed.
### Pred Gameplay:
- Adding Predator Soul - We want to bring Predators in line with what we feel is soulful gameplay. To that end we’re adding 15% bullet resistance and 20% slashing resistance to give Predators some much needed staying power on the battlefield.  Additionally Predator melee weapons will be rebalanced with an increase of 10 points damage to give them a better gamefeel.
- Placement Duels - Bi-monthly a dueling tournament will be held within each clan for the rank of Elder.  Winners will be given the Elder title and command of the Clan until the next tournament.
### Maps:
- New Map - Boxstation
- New Map - Shivas Snowball (Summer)
- New Map - Solaris Grassy Knoll
- New Map - Old Varadero
- New Map Feature - The dam on Trijent Dam can now be blown up to flood the map.
- New Map Feature - Shivas can now generate randomized ice formations on the surface.
- Random Map Event - Cross-over Rift: A rift spawns near the Almayer! From it pours things from another universe! Current list of potential crossovers include: Doctor Who, Star Trek, He-Man, and The Last Starfighter.
- Almayer Orientation -  To help each round feel different, we’ve used mirror tools to make the facing of the USS Almayer randomized in N/W/S/E orientation as well as horizontal and/or vertical mirroring.
- Communications Update - Due to the fluctuation in gameplay between static and mobile comms, we've decided to remove them all together and all planet side communications are done via RTO.
- “Open Req” Policy - To remove the historical problem of having no one run Req, the doors are open and marines can just grab what they need.
- Light Bulbs - Light bulbs now have a natural life span and will burn out in time.
- Experimental Round Mechanic - Rather than starting as marines, all USCM players start as survivors on “greenshift” until xeno infestation occurs and 3 or fewer survivors remain.  After this state, the standard USCM round begins.  To prevent griefing of this new round, all weapons are locked at the start, and craftable weapons cannot be made until the “Evil-ution” phase is high enough.
- Cat Update - Given that Jones would be dead in this time period, we are renaming Jones to Stephen.
- Whiskey Outpost - Given the popularity of the mode, a WO vote will be included at every round end.

### CODE REVIEW

**Comment by Doom-external** [comment]
i forget its april fools in australia already

**Comment by AGasCanister** [comment]
Of course the April Fools PR would have some actual good features!!

**Comment by DylonBR** [comment]
merge when?

**Comment by zzurge** [comment]
rofl

**Comment by PhantomEpicness** [comment]
missed an opportunity to say "The Future and Byond"

L  + RATIO + GBP DEDUCTED

File: code/CM 2.0/init.dm [hunk 1]
```diff
@@ -0,0 +1,13829 @@
+/*
+I mean, who actually reads code anyway?
```
Comment (Al-1ce) [inline code/CM 2.0/init.dm:2]: lol

**Comment by 567Turtle** [comment]
You guys are early

**Comment by blackcrystall** [comment]
*claps, give me one please

File: code/CM 2.0/2.0 Core Systems Rework.dm [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+Nam
+gravida
+erat
+nec
+nisi
+pellentesque
+elementum.
+
+
+*/
```
[the remark this answers is not in the document]
Comment (blackcrystall) [reply code/CM 2.0/2.0 Core Systems Rework.dm:437]: No, you need TAB at end!

File: code/CM 2.0/2.0 Mapping Systems Rework.dm [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+Nam
+gravida
+erat
+nec
+nisi
+pellentesque
+elementum.
+
+
+*/
```
[the remark this answers is not in the document]
Comment (blackcrystall) [reply code/CM 2.0/2.0 Mapping Systems Rework.dm:437]: Lets review that code, i think there some errors!

File: code/CM 2.0/2.0 Marine Rework.dm [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+Nam
+gravida
+erat
+nec
+nisi
+pellentesque
+elementum.
+
+
+*/
```
[the remark this answers is not in the document]
Comment (blackcrystall) [reply code/CM 2.0/2.0 Marine Rework.dm:437]: need rework that thing

**Review by blackdragonTOW — COMMENTED** [review]
Added some extra to the end that was needed

**Review by blackdragonTOW — COMMENTED** [review]
FOR REAL THIS TIME

**Comment by fira** [comment]
Due to the scope of the changes and some tweaks that might affect overall game reliability (for example, Xeno Voice Chat increases chances to get DDoSed substantially, reducing game stability), do note that these will be tested on the chinese staging server (United Country Royal Naval Marine Corps) prior to considering merging in mainline.

In the meanwhile follow established protocol for player-controversed changes by placing random tags and bugging author about GBP while discarding any opinions.

And please for the love of all things holy doublecheck wether you're on shadow closed source gitlab or github before pressing enter!

**Review by blackdragonTOW — COMMENTED** [review]
Man its like I really dont do this shit.

**Review by blackdragonTOW — COMMENTED** [review]
Wow, extra spaces before a commented out word.  Really? This is what you're hung up on?

**Review by blackdragonTOW — COMMENTED** [review]
Spaces!

**Review by blackdragonTOW — COMMENTED** [review]
SPACES

**Review by blackdragonTOW — COMMENTED** [review]
Did you know you had to include things? wild huh?

**Review by blackdragonTOW — COMMENTED** [review]
ALPHABETICAL ORDER?!?!

**Comment by Johannes2262** [comment]
As proud more-or-less MP main, I want the MP community program merged NOW. After banning all the admemes, nobody will be able to prevent me to ban everyone for shoving me per accident! As more-or-less CL main, I want the cocaine briefcase merged too.

**Comment by Nanu308** [comment]
based april fools moment

**Pull request closed without merging**